### ISSUE TITLE
How to save best fitted model when we use early stopping?

### ISSUE THREAD

**Issue by mohammad-saber** [deepchem/deepchem#2668]
## ❓ Questions & Help

Hi,
I am using "early stopping" for a GraphConvModel model.

```
import deepchem as dc

tasks, datasets, transformers = dc.molnet.load_delaney(featurizer='GraphConv')
train_dataset, valid_dataset, test_dataset = datasets

metric = dc.metrics.Metric(dc.metrics.r2_score)


# Early Stopping
model = dc.models.GraphConvModel(n_tasks=len(tasks), mode='regression', dropout=0.2,
                                 learning_rate=0.0001, model_dir='./result')

callback = dc.models.ValidationCallback(valid_dataset, 100, metric)
model.fit(train_dataset, nb_epoch=50, callbacks=callback)
```

Q1) After running the above code, what I get as `model`, is it the  best fitted model (model with the best value of metric for validation dataset)?

Q2) After fitting, folder `model_dir` contains several checkpoints like: `checkpoint, ckpt-2.index, ckpt-2.data-00000-of-00001`.
I was wondering which one is the best model?

Q3) How can I reload the best model only for prediction?

Thank you for your help.

**Comment by rbharath** [comment]
Could you come by the DeepChem office hours one week? https://forum.deepchem.io/t/announcing-the-deepchem-office-hours/293. Might be easiest to just discuss directly on a call!

**Comment by mohammad-saber** [comment]
Thank you for the invitation. Because of the time zone difference, a little difficult for me to join. However, I try to join the meeting.

**Comment by iceplussss** [comment]
`callback = dc.models.ValidationCallback(valid_dataset, 100, metric)`
->
`callback = dc.models.ValidationCallback(valid_dataset, 100, [metric])`
